pulp_ansible/app/galaxy/v1/views.py

```diff
@@ -0,0 +1,136 @@
+        roles = Role.objects.distinct("namespace", "name").filter(pk__in=distro_content)
+
+        namespace = self.request.query_params.get("owner__username", None)
+        if namespace:
+            roles = roles.filter(namespace__iexact=namespace)
+        name = self.request.query_params.get("name", None)
+        if name:
+            roles = roles.filter(name__iexact=name)
+
+        return roles
+
+    def get(self, request):
+        """
+        Get a single legacy role import task.
+        """
+        task_id = int(request.GET.get("id", None))
+
+        this_task = None
+        for t in Task.objects.all():
+            tid = str(t.pulp_id)
+            thash = abs(hash(tid))
+            if thash == task_id:
+                this_task = t
+                break
+
+        pulp_state = this_task.state.upper()
+        mtype = TYPE_MAP.get(pulp_state, pulp_state)
+        msg = ""
+        if this_task.error:
+            if this_task.error.get("traceback"):
+                tb = this_task.error["description"]
+                tb += "\n"
+                tb += this_task.error["traceback"]
+                msg += tb
+
+        v1state = STATE_MAP.get(pulp_state, pulp_state)
+        if v1state == "SUCCESS":
+            msg = "role imported successfully"
+        elif v1state == "RUNNING":
+            msg = "running"
+
+        return Response(
+            {
+                "results": [
+                    {
+                        "state": v1state,
+                        "id": task_id,
+                        "summary_fields": {
+                            "task_messages": [
+                                {
+                                    "id": datetime.datetime.now().isoformat(),
+                                    "message_text": msg,
+                                    "message_type": mtype,
+                                    "state": v1state,
+                                }
+                            ]
+                        },
+                    }
+                ]
+            }
+        )
+
+    def post(self, request):
+        """
+        Start a legacy role import task.
+        """
+        legacy_repo = AnsibleRepository.objects.filter(name=LEGACY_REPOSITORY_NAME).first()
+        kwargs = {
+            "ansible_repository": legacy_repo.pulp_id,
+            "github_user": request.data.get("github_user"),
+            "github_repo": request.data.get("github_repo"),
+            "github_reference": request.data.get("github_reference", ""),
+            "repository_id": request.data.get("repository_id"),
+            "alternate_role_name": request.data.get("alternate_role_name"),
+        }
+        role_name = kwargs["alternate_role_name"] or kwargs["github_repo"].replace(
+            "ansible-role-", ""
+        )
+        task = dispatch(legacy_role_import, exclusive_resources=[legacy_repo], kwargs=kwargs)
+        hashed = abs(hash(str(task.pulp_id)))
```

Comment:
Using `hash` I think is a good way to work around the integer versus UUID issue.